lib/cvss.js

```diff
@@ -150,6 +150,69 @@ function CVSS(vector) {
     throw new Error("The vector format is not valid!");
   }
 
+  /**
+   * Returns a Impact sub score
+   *
+   * 𝐼𝑆𝐶𝐵𝑎𝑠𝑒 = 1 − [(1 − 𝐼𝑚𝑝𝑎𝑐𝑡𝐶𝑜𝑛𝑓) × (1 − 𝐼𝑚𝑝𝑎𝑐𝑡𝐼𝑛𝑡𝑒𝑔) × (1 − 𝐼𝑚𝑝𝑎𝑐𝑡𝐴𝑣𝑎𝑖𝑙)]
+   *
+   * Scope Unchanged 6.42 × 𝐼𝑆𝐶Base
+   * Scope Changed 7.52 × [𝐼𝑆𝐶𝐵𝑎𝑠𝑒 − 0.029] − 3.25 × [𝐼𝑆𝐶𝐵𝑎𝑠𝑒 − 0.02]15
+   *
+   * @param {String} vector
+   * @returns {Number} Impact sub score
+   */
+  function getImpactSubScore() {
+    const vectorObject = util.getVectorObject(vector);
+    const C = util.findMetricValue("C", vectorObject).numerical;
+    const I = util.findMetricValue("I", vectorObject).numerical;
+    const A = util.findMetricValue("A", vectorObject).numerical;
+    const {S} = vectorObject;
+
+    // Calculate the impact using the formula from the CVSS v3.0 Specification Document
+    const impact = 1 - (1 - C) * (1 - I) * (1 - A);
+
+    // Check if the impact equal 0
+    if (impact === 0) return impact;
+
+    // If unchanged scope, multiply the impact by 6.42
+    let result = (6.42 * impact).toFixed(1);
+
+
+    // Check if the scope is changed
+    if (S === "C") {
+      // Scope Changed 7.52 × [𝐼𝑆𝐶𝐵𝑎𝑠𝑒 − 0.029] − 3.25 × [𝐼𝑆𝐶𝐵𝑎𝑠𝑒 − 0.02] ** 15
+      result = (7.52 * (impact - 0.029) - 3.25 * (impact - 0.02) ** 15).toFixed(1);
+    }
+
+    return Number(result);
```

Comment:
This implementation already exists:

https://github.com/turingsecure/cvss.js/blob/main/lib/score.js#L95

I'd suggest you just use that function here instead of your own implementation